shared/dev/dev-menu.native.js

```diff
@@ -37,7 +37,7 @@ class DevMenu extends Component {
         this.props.routeAppend('developer')
       }},
       {name: 'Search', hasChildren: true, onClick: () => {
-        this.props.pushNewSearch()
+        this.props.routeAppend('search')
```

Comment:
This fix needs to happen on desktop as well (I believe)
<img width="610" alt="screen shot 2016-06-22 at 5 11 02 pm" src="https://cloud.githubusercontent.com/assets/1152104/16313325/98d7882c-392d-11e6-91ae-05b5df07f61b.png">